src/notification.c

```diff
@@ -671,8 +671,11 @@ void notification_update_text_to_render(struct notification *n)
         }
 
         /* print age */
+        /* Note that this floors the notification's timestamp to the second.
+         * This ensures that all notifications ages will change at once, and
+         * that only one update will occur each second for this purpose. */
         gint64 hours, minutes, seconds;
-        gint64 t_delta = time_monotonic_now() - n->timestamp;
+        gint64 t_delta = time_monotonic_now() - (n->timestamp - n->timestamp % S2US(1));
```

Comment:
I don't think this one is neccesary. Please move the comment to queues_get_next_datachange, since that's where the magic happens.

Reply:
I actually believe that it is still necessary -- although the comment should be moved. If not, if a notification was issued at timestamp 0.5 and `show_age_threshold` is 0, and some other notification requires update (eg. expires) at timestamp 2.6, this will result in a strange outcome for the user:

* at t=0.5, the notification is shown with age=0s
* at t=1 (age update), the age is still shown as 0s
* at t=2 (age update), the age shows 1s — 0s was displayed for 1.5s, but this is not a huge problem as this must be bootstraped anyway
* at t=2.6, the age shows 2s — 1s was displayed for 0.6s, which will look weird to the user
* at t=3s, still 2s
* at t=4s, 3s — 2s displayed for 1.4s

This can be repeated with any number of unrelated updates, and can result in a weird display. I believe the best way to avoid this is to keep this patch; this way, the first second may be stretched longer than 1s, but all subsequent updates will be regular.

Reply:
Okay, that makes sense. It sounds like reasonable behavior.